

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>KIRP PE2 analysis &mdash; PE2 0.0.1 documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/theme_overrides.css" type="text/css" />

  
  
    <link rel="shortcut icon" href="../_static/logo.png"/>
  

  
  

  
    <link rel="canonical" href="https://arianemora.github.io/KIRP_PE2/examples/KIRP_type2.html" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script src="../_static/custom.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    

    <link rel="author" title="About these documents" href="../about.html" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="KIRP PE2 low FH" href="KIRP_type2_FHdeficient.html" />
    <link rel="prev" title="Data processing and download for DNA methylation and RNA" href="Preprocessing.html" />
 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html">
          

          
            
            <img src="../_static/logo.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                0.0.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">Package info</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../about.html">Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../about.html#data">Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../about.html#figures">Figures</a></li>
<li class="toctree-l1"><a class="reference internal" href="../references.html">References</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Reproducibility</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="Preprocessing.html">Data processing and download for DNA methylation and RNA</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">KIRP PE2 analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#starting-the-deseq2-analysis">Starting the DEseq2 Analysis</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#prefiltering-data">Prefiltering data</a></li>
<li class="toctree-l3"><a class="reference internal" href="#normalising-data-for-exploration">Normalising data for exploration</a></li>
<li class="toctree-l3"><a class="reference internal" href="#visualising-of-the-data">Visualising of the data</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#clustermap">Clustermap</a></li>
<li class="toctree-l4"><a class="reference internal" href="#pca">PCA</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#running-de-analysis">Running DE analysis</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#results-columns-explained">Results columns explained</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#having-a-look-at-the-distribution-of-specific-genes">Having a look at the distribution of specific genes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#or-just-look-at-the-raw-data">Or just look at the raw data</a></li>
<li class="toctree-l3"><a class="reference internal" href="#saving-the-data">Saving the data</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="KIRP_type2_FHdeficient.html">KIRP PE2 low FH</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">About</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="../changelog.html">Change log</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">PE2</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content style-external-links">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>KIRP PE2 analysis</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            
              <a href="https://github.com/arianemora/KIRP_PE2/blob/main/docs_src/src/examples/KIRP_type2.md" class="fa fa-github"> Edit on GitHub</a>
            
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="kirp-pe2-analysis">
<h1>KIRP PE2 analysis<a class="headerlink" href="#kirp-pe2-analysis" title="Permalink to this headline">¶</a></h1>
<p>Here we load the KIRP  PE2 data and perform a DE analysis and also look at counts for certain genes.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>&quot; If you need to install uncomment below&quot;
#BiocManager::install(&#39;DESeq2&#39;)
#BiocManager::install(&#39;EnhancedVolcano&#39;)
#install.packages(&quot;tidyverse&quot;)
#install.packages(&quot;stringr&quot;)

library(&quot;tidyverse&quot;)
library(&quot;DESeq2&quot;)
library(&quot;stringr&quot;)
library(EnhancedVolcano)

projectDir &lt;- &#39;data/&#39;

filename &lt;- paste(projectDir, &#39;TCGA_kidney_counts_KIRP-PE2-only_13082020.csv&#39;, sep=&#39;&#39;)

# Read in the count matrix
counts &lt;- read.csv(filename, header = TRUE, sep = &quot;,&quot;)
# We want to rename the rows and make sure they are labeled by the gene_id (note you&#39;ll have to select whatever your ID column name is here)
&quot; Set &lt;gene_id&gt; to be the column with the gene IDs &quot;
rownames(counts) &lt;- counts$gene_id
geneNames &lt;- counts$gene_name
geneIds &lt;- counts$gene_id

# select only the PE2 data
counts &lt;- dplyr::select(counts,contains(&quot;PE2&quot;))

# Here we get the names of the columns, for my stuff I always have all the info in one string as I find it makes it easier
# this means each of the &quot;groups&quot; are separated by a &quot;_&quot; this may be different for you
sampleNames &lt;- colnames(counts) # Sample names

# Lets get the names of the different factors
splitSampleNames &lt;- str_split(sampleNames, &#39;_&#39;)
project &lt;- sapply(splitSampleNames, function(x){x[2]})
sampleType &lt;- sapply(splitSampleNames, function(x){x[3]})
gender &lt;- sapply(splitSampleNames, function(x){x[4]})
race &lt;- sapply(splitSampleNames, function(x){x[5]})
tumourStage &lt;- sapply(splitSampleNames, function(x){x[6]})

# For DEseq2 we need to turn this into a dataframe
sampleDF = data.frame(project = project, sampleType = sampleType,
                      gender = gender, race = race, tumourStage = tumourStage)

# Make sure we don&#39;t include the ID in our columns
countMatrix &lt;- as.matrix(counts)

# We now set the row names to be the gene IDs
rownames(countMatrix) &lt;- geneNames

# Remove Nans
countMatrix[is.nan(countMatrix)] &lt;- 0
</pre></div>
</div>
<div class="section" id="starting-the-deseq2-analysis">
<h2>Starting the DEseq2 Analysis<a class="headerlink" href="#starting-the-deseq2-analysis" title="Permalink to this headline">¶</a></h2>
<p>Now we can start to run the DEseq2 analysis or the EdgeR, both use the data in the above format.</p>
<p>First we want to set the design for the experiment, from the above webpage:</p>
<blockquote>
<div><p>If the research aim is to determine for which genes the effect of treatment is different across groups, then interaction terms can be included and tested using a design such as ~ group + treatment + group:treatment.</p>
</div></blockquote>
<p>For us since we’re looking at TCGA maybe we want to look at our treatment as sampleTypes e.g. “Tumour” vs “PrimaryTissueNormal” and our groups as gender e.g: “female”, “male”.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Create a DEseq2 dataset from the counts using our count matrix note we need to set our colum data as factors</span>
<span class="n">ddsMat</span> <span class="o">&lt;-</span> <span class="n">DESeqDataSetFromMatrix</span><span class="p">(</span><span class="n">countData</span> <span class="o">=</span> <span class="n">countMatrix</span><span class="p">,</span>
                                 <span class="n">colData</span> <span class="o">=</span> <span class="n">sampleDF</span><span class="p">,</span>
                                 <span class="n">design</span> <span class="o">=</span> <span class="o">~</span> <span class="n">sampleType</span><span class="p">)</span>
</pre></div>
</div>
<div class="section" id="prefiltering-data">
<h3>Prefiltering data<a class="headerlink" href="#prefiltering-data" title="Permalink to this headline">¶</a></h3>
<p>This step is one of the most important - here we decide how many of the rows to keep. Since I know the TCGA data has many samples,
I am going to be more strict and only keep genes that have &gt; 5 counts in at least 10 samples. Something less stringent may be to just remove rows that have no counts.</p>
<p>Note the more samples that are kept, the more statistical tests you need to perform, thus they will need to pass a higher FDR. It may be important here to filter many of the not so good genes if you wish to pick up on smaller changes such as transcription factors, or more variable genes such as cell cycle genes.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="s2">&quot;</span>
<span class="n">Here</span> <span class="ow">is</span> <span class="n">the</span> <span class="n">example</span> <span class="n">of</span> <span class="n">something</span> <span class="n">you</span> <span class="n">might</span> <span class="n">do</span> <span class="k">if</span> <span class="n">you</span> <span class="n">have</span> <span class="n">only</span> <span class="n">very</span> <span class="n">few</span> <span class="n">samples</span><span class="p">:</span>

<span class="n">keep</span> <span class="o">&lt;-</span> <span class="n">rowSums</span><span class="p">(</span><span class="n">counts</span><span class="p">(</span><span class="n">dds</span><span class="p">))</span> <span class="o">&gt;</span> <span class="mi">1</span>
<span class="n">dds</span> <span class="o">&lt;-</span> <span class="n">dds</span><span class="p">[</span><span class="n">keep</span><span class="p">,]</span>
<span class="s2">&quot;</span>

<span class="c1"># Filter out genes with very few counts - here since our normal group have 10 samples we&#39;re being pretty conservative with 8</span>
<span class="n">numSamplesMeetingCriteria</span> <span class="o">&lt;-</span> <span class="mi">8</span>
<span class="n">numCountsInGene</span> <span class="o">&lt;-</span> <span class="mi">2</span>
<span class="n">keep</span> <span class="o">&lt;-</span> <span class="n">rowSums</span><span class="p">(</span><span class="n">counts</span><span class="p">(</span><span class="n">ddsMat</span><span class="p">)</span> <span class="o">&gt;=</span> <span class="n">numCountsInGene</span><span class="p">)</span> <span class="o">&gt;=</span> <span class="n">numSamplesMeetingCriteria</span>
<span class="n">dds</span> <span class="o">&lt;-</span> <span class="n">ddsMat</span><span class="p">[</span><span class="n">keep</span><span class="p">,]</span>

<span class="c1"># Let&#39;s print the number of rows</span>
<span class="n">nrow</span><span class="p">(</span><span class="n">dds</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="normalising-data-for-exploration">
<h3>Normalising data for exploration<a class="headerlink" href="#normalising-data-for-exploration" title="Permalink to this headline">¶</a></h3>
<p>While we don’t need to normalise the data for performing differential expression, it is good to normalise it for visualisation, e.g. when we do a PCA. This is because PCA aims to maximise the captured variance which without normalisation will capture the genes with the highest counts. The tutorial recommends either <em>Variance stabilising transformation</em> (VST) (for medium to large datasets e.g. n &gt; 30) or <em>rlog</em> (small datasets). These are preferred over just taking the <em>log transform</em> as in the log transform small count variances are over exemplified. I will just use VST since I have large datasets.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># blind = FALSE means that the transform is not &quot;unsupervised&quot; it is made aware of the design we set earlier in the dds</span>
<span class="n">vsd</span> <span class="o">&lt;-</span> <span class="n">vst</span><span class="p">(</span><span class="n">dds</span><span class="p">,</span> <span class="n">blind</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="visualising-of-the-data">
<h3>Visualising of the data<a class="headerlink" href="#visualising-of-the-data" title="Permalink to this headline">¶</a></h3>
<p>We’ll perform two visualisations:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mi">1</span><span class="p">)</span> <span class="n">clustermap</span> <span class="n">of</span> <span class="n">the</span> <span class="n">distance</span> <span class="n">between</span> <span class="n">samples</span>
<span class="mi">2</span><span class="p">)</span> <span class="n">PCA</span> <span class="n">of</span> <span class="n">the</span> <span class="n">samples</span>
</pre></div>
</div>
<p>For both of these we’ll use the normalised data (by VST).</p>
<div class="section" id="clustermap">
<h4>Clustermap<a class="headerlink" href="#clustermap" title="Permalink to this headline">¶</a></h4>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>library(&quot;pheatmap&quot;)
library(&quot;RColorBrewer&quot;)

# We calculate the euclidean distances between the samples. Note we need to transpose the matrix (t(assay(vsd))
sampleDists &lt;- dist(t(assay(vsd)))

sampleDistMatrix &lt;- as.matrix( sampleDists )
# Create some row names
rownames(sampleDistMatrix) &lt;- paste( vsd$sampleType, sep = &quot; - &quot; )
# Set the column names to be empty since these are just the same as the row names
colnames(sampleDistMatrix) &lt;- NULL
colors &lt;- colorRampPalette( rev(brewer.pal(9, &quot;Blues&quot;)) )(255)
anno &lt;- as.data.frame(colData(vsd)[, c(&quot;sampleType&quot;)])

pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         title=&quot;Distances between samples&quot;,
         col = colors)
</pre></div>
</div>
</div>
<div class="section" id="pca">
<h4>PCA<a class="headerlink" href="#pca" title="Permalink to this headline">¶</a></h4>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>
<span class="n">pcaData</span> <span class="o">&lt;-</span> <span class="n">plotPCA</span><span class="p">(</span><span class="n">vsd</span><span class="p">,</span> <span class="n">intgroup</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span> <span class="s2">&quot;sampleType&quot;</span><span class="p">),</span> <span class="n">returnData</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)</span>
<span class="n">percentVar</span> <span class="o">&lt;-</span> <span class="nb">round</span><span class="p">(</span><span class="mi">100</span> <span class="o">*</span> <span class="n">attr</span><span class="p">(</span><span class="n">pcaData</span><span class="p">,</span> <span class="s2">&quot;percentVar&quot;</span><span class="p">))</span>
<span class="n">ggplot</span><span class="p">(</span><span class="n">pcaData</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">PC1</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">PC2</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="n">sampleType</span><span class="p">,</span> <span class="n">shape</span> <span class="o">=</span> <span class="n">gender</span><span class="p">))</span> <span class="o">+</span>
  <span class="n">geom_point</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span><span class="mi">3</span><span class="p">)</span> <span class="o">+</span>
  <span class="n">xlab</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="s2">&quot;PC1: &quot;</span><span class="p">,</span> <span class="n">percentVar</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="s2">&quot;% variance&quot;</span><span class="p">))</span> <span class="o">+</span>
  <span class="n">ylab</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="s2">&quot;PC2: &quot;</span><span class="p">,</span> <span class="n">percentVar</span><span class="p">[</span><span class="mi">2</span><span class="p">],</span> <span class="s2">&quot;% variance&quot;</span><span class="p">))</span> <span class="o">+</span>
  <span class="n">coord_fixed</span><span class="p">()</span> <span class="o">+</span>
  <span class="n">ggtitle</span><span class="p">(</span><span class="s2">&quot;PCA with VST data&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="running-de-analysis">
<h3>Running DE analysis<a class="headerlink" href="#running-de-analysis" title="Permalink to this headline">¶</a></h3>
<p>Now we’ll actually run the DEseq analysis using what we designed before.</p>
<div class="section" id="results-columns-explained">
<h4>Results columns explained<a class="headerlink" href="#results-columns-explained" title="Permalink to this headline">¶</a></h4>
<blockquote>
<div><p>baseMean: average of the normalized count values, divided by the size factors, taken over all samples in the DESeqDataSet
log2FoldChange: effect size estimate e.g. how much a gene’s expression has changed as a result of the treatment (or in our case tumour)
lfcSE: uncertainty (standard error estimate) for the log2FoldChange
pvalue: the likelihood that the effect observed occured by chance (e.g. by experimetal variability etc)</p>
</div></blockquote>
<p>Note:</p>
<ol class="simple">
<li><p>alpha = p-value i.e. significance        –&gt; usually 0.05 (will have high FDR)</p></li>
<li><p>padj = FDR adjusted p values (by BH)     –&gt; usually 0.05 or 0.1</p></li>
<li><p>lfcThreshold = log fold change threshold –&gt; usually 1.0</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>
# Run DEseq2
dds &lt;- DESeq(dds)

# Build results table
res &lt;- results(dds)

# Sumarise the results
summary(res)

# Lastly, we may want to see the results of the high logfoldchange e.g. &gt; 1 with a padj value &lt; 0.05
table(res$padj &lt; 0.05)
</pre></div>
</div>
</div>
</div>
<div class="section" id="having-a-look-at-the-distribution-of-specific-genes">
<h3>Having a look at the distribution of specific genes<a class="headerlink" href="#having-a-look-at-the-distribution-of-specific-genes" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="c1"># Basic violin plot we need to normalise the data before doing this (we&#39;ll use edge R) Could alternativly just take log2 + 1</span>

<span class="c1">## Normalise just using log2 counts</span>

<span class="n">log2Counts</span> <span class="o">&lt;-</span> <span class="n">log2</span><span class="p">(</span><span class="n">countMatrix</span> <span class="o">+</span> <span class="mi">1</span><span class="p">)</span>

<span class="s2">&quot; Get the genes of interest &quot;</span>
<span class="n">geneOfInterest</span> <span class="o">&lt;-</span> <span class="n">c</span><span class="p">(</span><span class="s1">&#39;PFKFB1&#39;</span> <span class="p">,</span><span class="s1">&#39;HIRA&#39;</span><span class="p">)</span>
<span class="n">geneRow</span> <span class="o">&lt;-</span> <span class="n">log2Counts</span><span class="p">[</span><span class="n">rownames</span><span class="p">(</span><span class="n">log2Counts</span><span class="p">)</span> <span class="o">%</span><span class="ow">in</span><span class="o">%</span> <span class="n">geneOfInterest</span><span class="p">,</span> <span class="p">]</span>
<span class="n">colnames</span><span class="p">(</span><span class="n">geneRow</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">sampleType</span>
<span class="n">geneRow</span> <span class="o">&lt;-</span> <span class="n">t</span><span class="p">(</span><span class="n">geneRow</span><span class="p">)</span>
<span class="n">geneRow</span> <span class="o">&lt;-</span> <span class="n">cbind</span><span class="p">(</span><span class="n">sampleType</span><span class="p">,</span> <span class="n">data</span><span class="o">.</span><span class="n">frame</span><span class="p">(</span><span class="n">geneRow</span><span class="p">,</span> <span class="n">row</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">NULL</span><span class="p">))</span>

<span class="c1"># Plot the normal and then the tumour for both rows</span>
<span class="c1"># We have 39 tumour and 9 normal (note there may be duplicates for some patients)</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="n">ggplot</span><span class="p">(</span><span class="n">geneRow</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">sampleType</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">HIRA</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="n">sampleType</span><span class="p">))</span> <span class="o">+</span>
  <span class="n">geom_violin</span><span class="p">()</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="n">p</span> <span class="o">+</span> <span class="n">geom_dotplot</span><span class="p">(</span><span class="n">binaxis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span> <span class="n">stackdir</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">dotsize</span><span class="o">=</span><span class="p">)</span> <span class="o">+</span> <span class="n">ggtitle</span><span class="p">(</span><span class="s2">&quot;Log2 + 1 counts HIRA&quot;</span><span class="p">)</span>
<span class="n">ggsave</span><span class="p">(</span>
  <span class="s2">&quot;Log2CountsHIRA.png&quot;</span><span class="p">,</span>
  <span class="n">plot</span> <span class="o">=</span> <span class="n">last_plot</span><span class="p">())</span>
<span class="c1"># Plot the normal and then the tumour for both rows</span>
<span class="c1"># We have 39 tumour and 9 normal (note there may be duplicates for some patients)</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="n">ggplot</span><span class="p">(</span><span class="n">geneRow</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">sampleType</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">PFKFB1</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="n">sampleType</span><span class="p">))</span> <span class="o">+</span>
  <span class="n">geom_violin</span><span class="p">()</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="n">p</span> <span class="o">+</span> <span class="n">geom_dotplot</span><span class="p">(</span><span class="n">binaxis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span> <span class="n">stackdir</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">dotsize</span><span class="o">=</span><span class="p">)</span> <span class="o">+</span> <span class="n">ggtitle</span><span class="p">(</span><span class="s2">&quot;Log 2 + 1 counts PFKFB1&quot;</span><span class="p">)</span>
<span class="n">ggsave</span><span class="p">(</span>
  <span class="s2">&quot;Log2CountsPFKFB1.png&quot;</span><span class="p">,</span>
  <span class="n">plot</span> <span class="o">=</span> <span class="n">last_plot</span><span class="p">())</span>
</pre></div>
</div>
</div>
<div class="section" id="or-just-look-at-the-raw-data">
<h3>Or just look at the raw data<a class="headerlink" href="#or-just-look-at-the-raw-data" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>
<span class="s2">&quot; Get the genes of interest &quot;</span>
<span class="n">geneOfInterest</span> <span class="o">&lt;-</span> <span class="n">c</span><span class="p">(</span><span class="s1">&#39;PFKFB1&#39;</span> <span class="p">,</span><span class="s1">&#39;HIRA&#39;</span><span class="p">)</span>
<span class="n">geneRow</span> <span class="o">&lt;-</span> <span class="n">countMatrix</span><span class="p">[</span><span class="n">rownames</span><span class="p">(</span><span class="n">countMatrix</span><span class="p">)</span> <span class="o">%</span><span class="ow">in</span><span class="o">%</span> <span class="n">geneOfInterest</span><span class="p">,</span> <span class="p">]</span>
<span class="n">colnames</span><span class="p">(</span><span class="n">geneRow</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">sampleType</span>
<span class="n">geneRow</span> <span class="o">&lt;-</span> <span class="n">t</span><span class="p">(</span><span class="n">geneRow</span><span class="p">)</span>
<span class="n">geneRow</span> <span class="o">&lt;-</span> <span class="n">cbind</span><span class="p">(</span><span class="n">sampleType</span><span class="p">,</span> <span class="n">data</span><span class="o">.</span><span class="n">frame</span><span class="p">(</span><span class="n">geneRow</span><span class="p">,</span> <span class="n">row</span><span class="o">.</span><span class="n">names</span><span class="o">=</span><span class="n">NULL</span><span class="p">))</span>

<span class="c1"># Plot the normal and then the tumour for both rows</span>
<span class="c1"># We have 39 tumour and 9 normal (note there may be duplicates for some patients)</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="n">ggplot</span><span class="p">(</span><span class="n">geneRow</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">sampleType</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">HIRA</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="n">sampleType</span><span class="p">))</span> <span class="o">+</span>
  <span class="n">geom_violin</span><span class="p">()</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="n">p</span> <span class="o">+</span> <span class="n">geom_dotplot</span><span class="p">(</span><span class="n">binaxis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span> <span class="n">stackdir</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">dotsize</span><span class="o">=</span><span class="p">)</span> <span class="o">+</span> <span class="n">ggtitle</span><span class="p">(</span><span class="s2">&quot;Raw counts HIRA&quot;</span><span class="p">)</span>
<span class="n">p</span>
<span class="n">ggsave</span><span class="p">(</span>
  <span class="s2">&quot;RawCountsHIRA.png&quot;</span><span class="p">,</span>
  <span class="n">plot</span> <span class="o">=</span> <span class="n">last_plot</span><span class="p">())</span>
<span class="c1"># Plot the normal and then the tumour for both rows</span>
<span class="c1"># We have 39 tumour and 9 normal (note there may be duplicates for some patients)</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="n">ggplot</span><span class="p">(</span><span class="n">geneRow</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">sampleType</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">PFKFB1</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="n">sampleType</span><span class="p">))</span> <span class="o">+</span>
  <span class="n">geom_violin</span><span class="p">()</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="n">p</span> <span class="o">+</span> <span class="n">geom_dotplot</span><span class="p">(</span><span class="n">binaxis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span> <span class="n">stackdir</span><span class="o">=</span><span class="s1">&#39;center&#39;</span><span class="p">,</span> <span class="n">dotsize</span><span class="o">=</span><span class="p">)</span> <span class="o">+</span> <span class="n">ggtitle</span><span class="p">(</span><span class="s2">&quot;Raw counts PFKFB1&quot;</span><span class="p">)</span>
<span class="n">p</span>
<span class="n">ggsave</span><span class="p">(</span>
  <span class="s2">&quot;RawCountsPFKFB1.png&quot;</span><span class="p">,</span>
  <span class="n">plot</span> <span class="o">=</span> <span class="n">last_plot</span><span class="p">())</span>
</pre></div>
</div>
</div>
<div class="section" id="saving-the-data">
<h3>Saving the data<a class="headerlink" href="#saving-the-data" title="Permalink to this headline">¶</a></h3>
<p>We just want to save the data to a CSV and use the gene names as well as the ensembl IDS</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span># Add gene names
res$symbol &lt;- rownames(dds)
resOrdered &lt;- res[order(res$padj),]
outputFilename &lt;- paste(projectDir, &#39;DEseq2_NormalVsTumour_KIRP-PE2.csv&#39;, sep=&#39;&#39;)
write.csv(resOrdered, file = outputFilename)
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span># Let&#39;s make a volcano plot
EnhancedVolcano(res,
  lab = res$symbol,
  x = &#39;log2FoldChange&#39;,
  y = &#39;padj&#39;,
  title = &#39;TCGA PE2 KIRP dataset&#39;,
  xlim = c(-5, 8),
  ylim = c(0, 15))
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>
# Top genes as per variance
topVarGenes &lt;- head(order(rowVars(assay(vsd)), decreasing = TRUE), 20)

mat  &lt;- assay(vsd)[ topVarGenes, ]
rownames(mat) &lt;- res$symbol[topVarGenes]
mat  &lt;- mat - rowMeans(mat)
anno &lt;- as.data.frame(colData(vsd)[, c(&quot;sampleType&quot;,&quot;gender&quot;, &quot;tumourStage&quot;)])
pheatmap(mat, annotation_col = anno, show_colnames = F, main=&quot;Top 20 most variable genes&quot;)
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span># Top genes as per significance
topVarGenes &lt;- head(order(res$padj, decreasing = FALSE), 20)

mat  &lt;- assay(vsd)[ topVarGenes, ]
rownames(mat) &lt;- res$symbol[topVarGenes]
mat  &lt;- mat - rowMeans(mat)
anno &lt;- as.data.frame(colData(vsd)[, c(&quot;sampleType&quot;,&quot;gender&quot;, &quot;tumourStage&quot;)])
pheatmap(mat, annotation_col = anno, show_colnames = F, main=&quot;Top 20 most significant genes&quot;)
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span># Top genes as per logFC
# Top genes as per significance
topVarGenes &lt;- head(order(abs(res$log2FoldChange), decreasing = TRUE), 20)
mat  &lt;- assay(vsd)[ topVarGenes, ]
rownames(mat) &lt;- res$symbol[topVarGenes]
mat  &lt;- mat - rowMeans(mat)
anno &lt;- as.data.frame(colData(vsd)[, c(&quot;sampleType&quot;,&quot;gender&quot;, &quot;tumourStage&quot;)])
pheatmap(mat, annotation_col = anno, show_colnames = F, main=&quot;Top 20 largest fold change&quot;)
</pre></div>
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="KIRP_type2_FHdeficient.html" class="btn btn-neutral float-right" title="KIRP PE2 low FH" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="Preprocessing.html" class="btn btn-neutral float-left" title="Data processing and download for DNA methylation and RNA" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2022, Ariane Mora.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>